### DISCUSSION TITLE
How to log QO100 QSO ?

### DISCUSSION THREAD

**Discussion by abarrau** [magicbug/Cloudlog#2491] [General]
Hello, 
Question to the community: how do you report QO100 QSOs in Cloudlog?

Initially I indicated for QSOs in SSB: 13cm (in TX), but with a frequency of 10Ghz,
Using GridTracker, and automatic sync for the FT8, it indicates: 3cm (in TX).

So, I have a doubt that the right mode to take... (to avoid having synchronization problems with LoTW or QRZ)

What is your opinion ?
THANKS,

Aurélien / F4ANS

**Comment by phl0** [comment]
Hi Aurélien,

thank you for the FT8 QSO on QO-100 just minutes ago. The correct way to log it is: FREQ / BAND set to 2.400MHz / 13cm. FREQ_RX / BAND_RX is 10.489MHz / 3cm. This is how you would get confirmation on LoTW. 

And just to let you know. There is a nice version of WSJT-X called WSJT-X improved which allows to log to Cloudlog directly and also supports SAT mode and RX frequency. You can find it here:

https://sourceforge.net/projects/wsjt-x-improved/files/WSJT-X_v2.7.1/

In Tests there is even a dev version which supports full duplex on satellites.

   vy73 de Florian DF2ET

**Reply by abarrau** [reply]
Hi Florian, 
Yes, thanks the FT8 QSO today :)
Ok for your answer, i will update info in bdd for all QO100 QSO :)

Good news for the new version. 
I just install the 2.7.1, but API flow is not ok for me when i test it "api invialid" ... :( ... while this same API key works correctly with GridTracker.

73, Aurélien

**Reply by phl0** [reply]
Which operating system are you using? And is your Cloudlog on HTTPS?

**Reply by abarrau** [reply]
Client OS : Win10 Pro
Cloudlog working on https

in fact, i use the same config with GridTracker and WSJTX2.7.1 (URI & API Key), but for WSJTX it's not ok. 
I'll look to see if there's a debug log to get more info. just the "test" button turns red

**Reply by phl0** [reply]
Check here:
https://github.com/magicbug/Cloudlog/wiki/WSJT-X-Integration#wsjt-x-improved-plus-versions-by-dg2ycb
You will need OpenSSL installed.

**Reply by abarrau** [reply]
Hi Flo, 
in fact, since the cloudlog new version 2.4.9, my api was broken. 
is it always OK for you ?

**Reply by phl0** [reply]
My API still works. Can you have a look at the logs please?

**Reply by abarrau** [reply]
Hi, 
Nothing in the log ... I add information for debug, and i found where i am stopped

```
if(!isset($obj['station_profile_id']) || $this->stations->check_station_against_user($obj['station_profile_id'], $userid) == false) {
	http_response_code(401);
	echo json_encode(['status' => 'failed', 'reason' => "station id does not belong to the API key owner."]);
	die();
}
```
->> I think it's this test (line 182 on controller/api.php file) is not good, because $obj['station_profile_id'] not exist.

```
obj=Array
(
    [key] => xxxxxxxxxxxxxxx
    [type] => adif
    [string] => <eor>
)
```
So i thinks it's concidering not authentified.

**Reply by magicbug** [reply]
My suggestion would be create a new API key see if it makes any difference

**Reply by abarrau** [reply]
Ok, i understood the problem. 
For WSJTX 2.7.1-devel --> it's ok ; API KEY is green now
For GridTracker --> 
- when i test from the logging setting page, "test button" return "invalid response" ... because "station_profile_id" was not send
- but, when i make a new QSO from wsjtx, (with cloudlog disabled, and send it by GridTracker), the QSO is correctly create ("station_profile_id" was send)

So all is ok now ...
The problem is on GridTracker software, not on Cloudlog !

Thanks

**Reply by abarrau** [reply]
Just a last question : on WSJTX 2.7.1-devel, i didn't see the QO-100 satellite on the list ... how can i add it ?

**Reply by magicbug** [reply]
OK, hopefully theres a new gridtracker version available that includes the station profile id

**Comment by phl0** [comment]
> Just a last question : on WSJTX 2.7.1-devel, i didn't see the QO-100 satellite on the list ... how can i add it ?

Probably need an updated sat.dat file. I scraped valid SATs from LoTW config and is available here: https://www.df2et.de/cqrlog/sat.dat

**Reply by abarrau** [reply]
All is ok now ;) 
Nice feature ! thanks
